### PR TITLE
Dependent app example

### PR DESCRIPTION
**Pull request by shilpabijam**
An example to check for two set up time dependencies - Weather app is installed and if its version is >=1.0. Its identical to the Weather app except for the setup time dependencies.

### CODE REVIEW

File: case_studies/weather_dependent_app_example/appserver/static/javascript/views/app.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+    async handleSubmit(event) {
+      event.preventDefault();
+
+      await Setup.perform(splunk_js_sdk, this.state)
+    }
+
+    render() {
+      return e("div", null, [
+        e("h2", null, "Check to see if dependent app is installed and has the required version. App configuration is not complete without it. If dependency requirements are satisfied, the app configuration will complete."),
```
Comment (tacole02) [inline case_studies/weather_dependent_app_example/appserver/static/javascript/views/app.js:39]: I suggest adjusting this text to explain exactly what the setup is checking for.

Something like:

"This app requires you to install the weather app in order to complete the setup process. If this dependency requirement is satisfied, then app setup completes and you are directed to the app home page. If this dependency requirement is not satisfied, then you are prompted to install the weather app."

File: case_studies/weather_dependent_app_example/README.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
+# Overview
+
+This is an example Splunk App to demonstrate how to use setup pages that adhere to Splunk Application Certification standards.
```
Comment (tacole02) [inline case_studies/weather_dependent_app_example/README.md:3]: I'd add a bit more context here:

"This is an example Splunk App that demonstrates how to use a setup page to check for the installation of app dependencies. This example adheres to Splunk Application Certification standards.

This app requires that users install the weather app as a dependency in order to complete the setup process. The setup page checks whether the dependent app is installed before allowing users to access the app home page."

File: case_studies/weather_dependent_app_example/README.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
+# Overview
```
Comment (tacole02) [inline case_studies/weather_dependent_app_example/README.md:1]: I'd change the app name to something more specific. Maybe "Dependency checking setup page app".

File: case_studies/weather_dependent_app_example/appserver/static/javascript/views/setup_page.js [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+        let configured = false
+        const dependent_app_name = "weather_app_example"
+
+        var apps = await Config.list_apps(splunk_js_sdk_service);
+        var current_app = apps.item(dependent_app_name);
+       
+        if (current_app.properties().configured == true) {
+            configured = true
+        } else {
+            alert('Error:' + 'Dependent weather_app_example is not installed!');
```
Comment (tacole02) [inline case_studies/weather_dependent_app_example/appserver/static/javascript/views/setup_page.js:38]: "The weather-app-example dependency is not installed. Install and configure the weather-app-example, then try again."

File: case_studies/weather_dependent_app_example/appserver/static/javascript/views/setup_page.js [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+        }
+        let version = parseInt(current_app.properties().version)
+
+        if (version < 1) {
+            alert('Error:' + 'Upgrade weather app version to atleast 1.0!');
```
Comment (tacole02) [inline case_studies/weather_dependent_app_example/appserver/static/javascript/views/setup_page.js:43]: "The weather-app-example dependency does not meet version requirements. Upgrade to version 1.0 or above, then try again."

File: case_studies/weather_dependent_app_example/appserver/static/javascript/views/app.js (hunk 1 above)
Comment (shilpabijam) [reply case_studies/weather_dependent_app_example/appserver/static/javascript/views/app.js:39]: Fixed.

File: case_studies/weather_dependent_app_example/README.md (hunk 2 above)
Comment (shilpabijam) [reply case_studies/weather_dependent_app_example/README.md:3]: Fixed.

File: case_studies/weather_dependent_app_example/README.md (hunk 3 above)
Comment (shilpabijam) [reply case_studies/weather_dependent_app_example/README.md:1]: Updated.

File: case_studies/weather_dependent_app_example/appserver/static/javascript/views/setup_page.js (hunk 4 above)
Comment (shilpabijam) [reply case_studies/weather_dependent_app_example/appserver/static/javascript/views/setup_page.js:38]: Updated.

File: case_studies/weather_dependent_app_example/appserver/static/javascript/views/setup_page.js (hunk 5 above)
Comment (shilpabijam) [reply case_studies/weather_dependent_app_example/appserver/static/javascript/views/setup_page.js:43]: Updated.

File: case_studies/dependency_checking_app_example/README.md [hunk 6]
```diff
@@ -0,0 +1,46 @@
+# Overview
+
+This is an example Splunk App that demonstrates how to use a setup page to check for the installation of app dependencies. This example adheres to Splunk Application Certification standards.
+
+This app requires that users configure the weather app as a dependency in order to complete the setup process. The setup page checks whether the dependent app is configured, and updated to specific version before allowing users to access the app home page."
```
Comment (tacole02) [inline case_studies/dependency_checking_app_example/README.md:5]: NIT: remove comma

Comment (tacole02) [inline case_studies/dependency_checking_app_example/README.md:5]: NIT: "updated to a specific version"

File: case_studies/dependency_checking_app_example/README.md [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+The dashboard view specifies its CSS and JavaScript resources and points to `appserver/static/javascript/setup_page.js`.
+
+And finally the `setup_page.js` imports a React app from `appserver/static/javascript/views/app.js`.
+
+There are two apps installed out-of-the-box when you `make` examples repo. Both not configured to begin with. Follow these steps to run through the dependency_checking_setup_page_app example:
+- First navigate to the dependency_checking_app and attempt to configure. 
+- There will be an alert that the `Weather_app_example` is not configured.
+- Navigate to the `weather_app_example` and finish set up configuration. Ensure the version installed is greater or equal to 1.0.
+- Navigate back to the dependency_checking_app and attempt to configure. 
+- Set up configuration should complete successfully.
```
Comment (tacole02) [inline case_studies/dependency_checking_app_example/README.md:26]: I think outlining the workflow here is a great idea :)

**Review by tacole02 — APPROVED** [review]

**Review by zenmoto — APPROVED** [review]

File: case_studies/dependency_checking_app_example/metadata/local.meta [hunk 8]
```diff
@@ -0,0 +1,8 @@
+[app/install]
```
[the remark this answers is not in the document]
Comment (shilpabijam) [reply case_studies/dependency_checking_app_example/metadata/local.meta]: Fixed.

File: case_studies/dependency_checking_app_example/metadata/default.meta [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+
+
+### VIEWSTATES: even normal users should be able to create shared viewstates
+
+[viewstates]
+access = read : [ * ], write : [ * ]
+export = system
+
+
+################################################################################
```
[the remark this answers is not in the document]
Comment (shilpabijam) [reply case_studies/dependency_checking_app_example/metadata/default.meta:38]: Fixed.

File: case_studies/dependency_checking_app_example/appserver/static/javascript/setup_page.js [hunk 10]
```diff
@@ -0,0 +1,39 @@
+"use strict";
+
+var app_name = "./dependency_checking_app_example";
```
[the remark this answers is not in the document]
Comment (shilpabijam) [reply case_studies/dependency_checking_app_example/appserver/static/javascript/setup_page.js]: Fixed.

File: case_studies/dependency_checking_app_example/appserver/static/javascript/views/util.js [hunk 11]
```diff
@@ -0,0 +1,26 @@
+function promisify(fn) {
+  console.log("promisify: Don't use this in production! Use a proper promisify library instead.")
```
[the remark this answers is not in the document]
Comment (shilpabijam) [reply case_studies/dependency_checking_app_example/appserver/static/javascript/views/util.js:2]: @zenmoto I have opened https://jira.splunk.com/browse/DVPL-9071 for this.

**Pull request merged**